Add-On UI/scripts/process_query.js

```diff
@@ -0,0 +1,236 @@
+  'dateRange': 'null',
+  'timeGranularity': 'null'
+}
+
+//function to set keys of JsonObj, called from client side javascript
+function createJsonObject(intent,tableData,metric,summaryOperator,isAsc,k,
+  dimensions,slices,dateRange,timeGranularity,outputSheet) {
+
+  /*
+   * set intent 
+   * JsonObj.intent is a string storing intent name
+   */
+  JsonObj.intent = intent;
+  
+  /*
+   * set table and rowRange
+   * JsonObj.table is a 2d array consisting of sheet's data
+   */
+  JsonObj.table = getTable(tableData);
+  
+  /*
+   * set metric
+   * JsonObj.metric is a string storing column name
+   */
+  JsonObj.metric = metric;
+  
+  /*
+   * set dimension
+   * JsonObj.dimensions is a 1d array of strings storing name of output columns
+   */
+  if(dimensions.length > 0)
+    JsonObj.dimensions = dimensions;
+  
+  /*
+   * set summary
+   * JsonObj.summaryOperator is a string storing the name of operation to be applied
+   */
+  JsonObj.summaryOperator = summaryOperator;
+  
+  /* 
+   * set sortOrder
+   * JsonObj.isAsc is a boolean variable which stores true for ascending order
+   */
+  JsonObj.isAsc = isAsc;
+  
+  /* 
+   * set limit
+   * JsonObj.k is an Integer storing K 
+   * (if user doesn't specify values of k -5000000 default)
+   */
+  JsonObj.k = k;
+   
+  /*
+   * set slice
+   * JsonObj.slices is a 1d array of objects
+   * each object has 3 keys - sliceCol, sliceOp, sliceVal
+   * sliceCol is a string, sliceOp is a string, sliceVal is 1d array of strings
+   */
+  if(slices.length > 0)
+    JsonObj.slices = slices;
+   
+  /*
+   * set dateRange
+   * JsonObj.dateRange is an object with 3 keys - dateCol, dateStart, dateEnd
+   * dateCol is a string, dateStart and dateEnd are dates (yyyy-mm-dd) 
+   */
+  if(dateRange.dateCol !== '' && dateRange.dateStart !== '' && dateRange.dateEnd !== '') {
+    dateRange.dateStart = new Date(dateRange.dateStart);
+    dateRange.dateEnd = new Date(dateRange.dateEnd);
+    JsonObj.dateRange=dateRange;
+    //formatDate(dateRange.dateCol);
+  }
+    
+  /*
+   * set timeGranularity
+   * JsonObj.timeGranularity is a string containing the selected value
+   */
+  JsonObj.timeGranularity = timeGranularity;
+  
+  //printing the JsonObj
+  Logger.log(JsonObj);
+
+  //printing JsonObj in json format
+  Logger.log(JSON.stringify(JsonObj));
+  
+  //calling GCP function
+  var outputTable = callGcpFunc();
+  
+  //converting date from json into javascript date object
+  outputTable = formatDate(outputTable);
+
+  //creating a new sheet with name outputSheet and outputTable as its contents
+  createNewSheet(outputTable,outputSheet);
+  
+}
+
+//function to get the table data and set rowRange of JsonObj
+function getTable(tableData) {
+
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var objRowRange = {
+      'header' : 1,
+      'rowStart': 2,
+      'rowEnd': 2
+  }
+
+  //if range is not specified we consider entire data 
+  //else we use the range provided
+
+  if(tableData[0] === '') {
```

Comment:
Less discuss offline in chat on what is exactly happening here.  how is  if(tableData[0] === '') checking that range is not specified, looks like i m missing something here.